src/data/vector2.lean

```diff
@@ -183,15 +183,19 @@ lemma remove_nth_insert_nth_ne {v : vector α (n+1)} :
     rcases lt_trichotomy i j with h | h | h,
     { have h_nji : ¬ j < i := lt_asymm h,
       have j_pos : 0 < j := lt_of_le_of_lt (zero_le i) h,
-      simp [h, h_nji, fin.lt_iff_val_lt_val, -subtype.val_eq_coe],
-      rw [show j.pred = j - 1, from rfl, list.insert_nth_remove_nth_of_ge, nat.sub_add_cancel j_pos],
+      rw [dif_neg], swap, { exact h_nji },
```

Comment:
Isn't this just `rw dif_neg h_nji`, and so on?

Reply:
@rwbarton  Nope, unfortunately not.

Reply:
`split_ifs` doesn't clean this up better, does it? This is pretty ugly :(